File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        poof = True if x > 0 else False
        x = abs(x)
        rev = 0
        
        while x > 0: 
            rev = (rev * 10) + (x % 10)
            x = x // 10
            
            
        if poof: 
            if(rev > ( ( 2**31 ) -1 ) ):
                return 0
        else:
            if(rev > ( 2**31  ) ):
                return 0
        
        return rev if poof else -1 * rev
        
        

File: 0007-reverse-integer/0007-reverse-integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut negative = false;
        if x < 0 { negative = true; }    
        let mut ans = x.abs().to_string();
        ans = ans.chars().rev().collect();
        
        let a = match ans.parse::<i32>() {
            Ok(n) => if negative {-n} else {n},
            Err(_) => 0
        };
        a
    }
}